pyproject.toml

```diff
@@ -152,25 +154,25 @@ packages = [
 
 [project.optional-dependencies]
 test = [
-    "pytest>=7.4.0,<8",
+    "pytest>=8,<9",
     "pytest-cov>=4.1.0,<5",
     "pytest-xdist>=3.3.1,<4",
     "GitPython>=3.1.0,<3.2",
-    "filelock>=3.12.3,<3.13",
+    "filelock>=3.15.1,<4",
     "requests",
     "requests-cache>=1.2.1,<2",
 ]
 
 lint = [
-    "isort==5.13.2",
     "mypy==1.14.1",
-    "black==23.12.0",
-    "flake8==6.1.0",
-    "flake8-bugbear==23.12.2",
-    "flake8-docstrings==1.7.0",
-    "flake8-spellcheck==0.28.0",
-    "flake8-unused-arguments==0.0.13",
+    "ruff==0.11.8",
     "vulture==2.14.0",
+    # Black and isort are taken care of by ruff
+    #    "black==23.12.0",
+    #    "isort==5.13.2",
+    # ruff does not cover spellcheck currently. It's an open issue.
+    # https://github.com/astral-sh/ruff/issues/5014
+    #    "flake8-spellcheck==0.28.0",
```

Comment:
Comments should describe the current state, while the commit message should be used to describe the reasoning behind changes.

```suggestion
```